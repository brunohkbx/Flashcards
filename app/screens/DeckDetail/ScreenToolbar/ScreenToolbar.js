import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Toolbar,
  ToolbarBackAction,
  ToolbarContent,
  ToolbarAction
} from 'react-native-paper';

class ScreenToolbar extends Component {
  navigateTo = route => {
    const { navigation } = this.props;
    const { deckId } = navigation.state.params;

    navigation.navigate(route, { deckId });
  }

  render() {
    const { navigation } = this.props;

    return (
      <Toolbar>
        <ToolbarBackAction onPress={() => navigation.goBack()} />
        <ToolbarContent title="Deck Details"/>
        <ToolbarAction
          icon="question-answer"
          onPress={() => {}}
        />
        <ToolbarAction
          icon="edit"
          onPress={() => this.navigateTo('EditDeck')}
        />
      </Toolbar>
    );
  }
}

ScreenToolbar.propTypes = {
  navigation: PropTypes.object.isRequired
};

export default ScreenToolbar;
