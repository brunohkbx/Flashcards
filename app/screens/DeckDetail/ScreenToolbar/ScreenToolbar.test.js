import React from 'react';
import { shallow } from 'enzyme';
import ScreenToolbar from './ScreenToolbar';

describe('ScreenToolbar', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      navigation: {
        goBack: jest.fn(),
        state: jest.fn(),
        navigate: jest.fn()
      }
    }, propOverrides);

    const wrapper = shallow(<ScreenToolbar {...defaultProps} />);
    const wrapperInstance = wrapper.instance();

    const goBack = () => wrapper.find('ToolbarBackAction').simulate('press');
    const edit = () => wrapper.find({ icon: 'edit' }).simulate('press');

    return { wrapper, wrapperInstance, goBack, edit };
  };

  it('renders properly', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  describe('navigateTo', () => {
    it('navigates to the given screen and pass deckId as props', () => {
      const mockNavigate = jest.fn();
      const { wrapperInstance } = setup();
      wrapperInstance.props.navigation.state.params = { deckId: '1' };
      wrapperInstance.props.navigation.navigate = mockNavigate;

      wrapperInstance.navigateTo('EditDeck');

      expect(mockNavigate).toHaveBeenCalledWith('EditDeck', { deckId: '1'} );
    });
  });

  test('ToolbarBackAction onPress goes to previous screen', () => {
    const mockNavigationGoBack = jest.fn();
    const { wrapperInstance, goBack } = setup();
    wrapperInstance.props.navigation.goBack = mockNavigationGoBack;

    goBack();

    expect(mockNavigationGoBack).toHaveBeenCalled();
  });

  test('ToolbarAction Edit calls navigateTo EditDeck', () => {
    const mockNavigateTo = jest.fn();
    const { wrapperInstance, edit } = setup();
    wrapperInstance.navigateTo = mockNavigateTo;

    edit();

    expect(mockNavigateTo).toHaveBeenCalled();
  });
});
