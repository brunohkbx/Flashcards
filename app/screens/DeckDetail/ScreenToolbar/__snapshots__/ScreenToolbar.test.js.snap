// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenToolbar renders properly 1`] = `
<withTheme(Appbar.Header)>
  <Appbar.BackAction
    accessibilityLabel="Back"
    onPress={[Function]}
  />
  <withTheme(Appbar.Content)
    title="Deck Details"
  />
  <Appbar.Action
    disabled={false}
    icon="question-answer"
    onPress={[Function]}
    size={24}
  />
  <Appbar.Action
    icon="edit"
    onPress={[Function]}
    size={24}
  />
</withTheme(Appbar.Header)>
`;

exports[`ScreenToolbar renders properly when quiz action is desabled 1`] = `
<withTheme(Appbar.Header)>
  <Appbar.BackAction
    accessibilityLabel="Back"
    onPress={[Function]}
  />
  <withTheme(Appbar.Content)
    title="Deck Details"
  />
  <Appbar.Action
    disabled={true}
    icon="question-answer"
    onPress={[Function]}
    size={24}
  />
  <Appbar.Action
    icon="edit"
    onPress={[Function]}
    size={24}
  />
</withTheme(Appbar.Header)>
`;
